src/Styleguide/Components/Lightbox/Lightbox.tsx

```diff
@@ -0,0 +1,286 @@
+          slider: {
+            ...this.state.slider,
+            min: this.state.viewer.viewport.getMinZoom(),
+            max: this.state.viewer.viewport.getMaxZoom(),
+            value: this.state.viewer.viewport.getHomeZoom(),
+          },
+        })
+      })
+    )
+    this.detectActivity()
+  }
+
+  show = event => {
+    this.setState({ shown: true, showZoomSlider: true })
+  }
+
+  hide = () => {
+    this.setState({ shown: false })
+    if (this.state.viewer) {
+      this.state.viewer.destroy()
+      this.state.viewer = null
+    }
+    document.removeEventListener(KEYBOARD_EVENT, this.handleKeyPress)
+    clearTimeout(this.state.activityTimer)
+  }
+
+  handleKeyPress = event => {
+    if (event && event.key === "Escape") {
+      this.hide()
+    }
+  }
+
+  detectActivity = throttle(() => {
+    clearTimeout(this.state.activityTimer)
+    this.setState({
+      showZoomSlider: true,
+      activityTimer: setTimeout(() => {
+        this.setState({
+          showZoomSlider: false,
+        })
+      }, HIDE_ZOOM_SLIDER_AFTER),
+    })
+  }, 500) as () => void
+
+  zoomBy = amount => {
+    if (this.state.viewer.viewport) {
+      this.state.viewer.viewport.zoomBy(amount)
+      this.state.viewer.viewport.applyConstraints()
+    }
+  }
+
+  zoomIn = () => {
+    this.zoomBy(ZOOM_PER_CLICK)
+  }
+
+  zoomOut = () => {
+    this.zoomBy(1 / ZOOM_PER_CLICK)
+  }
+
+  initSeaDragon = () => {
+    this.state.promisedDragon.then(OpenSeaDragon => {
+      const viewer = OpenSeaDragon({
+        element: this.state.deepZoomRef.current,
+
+        debugMode: false,
+        showNavigationControl: false,
+        immediateRender: false,
+        useCanvas: true,
+        constrainDuringPan: false,
+        blendTime: 0.0,
+        animationTime: 1.5,
+        springStiffness: 15.0,
+        maxZoomPixelRatio: 1.0,
+        minZoomImageRatio: 0.9,
+        zoomPerClick: ZOOM_PER_CLICK,
+        zoomPerScroll: 1.4,
+        clickDistThreshold: 5,
+        clickTimeThreshold: 300,
+        visibilityRatio: 1,
+        tileSources: this.props.deepZoom,
+
+        gestureSettingsTouch: {
+          scrolltozoom: false,
+          clicktozoom: true,
+          pinchtozoom: true,
+          flickenabled: true,
+          flickminspeed: 20,
+          flickmomentum: 0.4,
+        },
+      })
+      document.addEventListener(KEYBOARD_EVENT, this.handleKeyPress)
+      this.setState({
+        viewer,
+      })
+    })
+  }
+
+  onSliderChanged = event => {
+    this.state.viewer.viewport.zoomTo(event.target.value)
+  }
+
+  onZoomChanged = () => {
+    if (!this.state.viewer) return
+    this.setState({
+      slider: {
+        ...this.state.slider,
+        min: this.state.viewer.viewport.getMinZoom(),
+        max: this.state.viewer.viewport.getMaxZoom(),
+        value: this.state.viewer.viewport.getZoom(),
+      },
+    })
+  }
+
+  componentDidMount() {
+    this.setState({
+      element: document.getElementById("lightbox-container"),
```

Comment:
Is it possible to not refer to a hardcoded ID here but let the user specify one instead?